src/main/src/main/java/org/geoserver/security/SecuredResourceNameChangeListener.java

```diff
@@ -50,7 +45,31 @@ public void handleAddEvent(CatalogAddEvent event) throws CatalogException {
 
     @Override
     public void handleRemoveEvent(CatalogRemoveEvent event) throws CatalogException {
-        // ignore
+        String resourceWorkSpaceName = null;
+        CatalogInfo resourceInfo = event.getSource();
+        String resourceName = null;
+        if (resourceInfo != null && resourceInfo instanceof LayerInfo) {
```

Comment:
If I'm not mistaken, ``resourceInfo instanceof LayerInfo`` will yield ``false`` if ``resourceInfo`` is NULL, hence no need the ``resourceInfo != null``check.